python/rbac/services/role.py

```diff
@@ -11,13 +11,35 @@
 # limitations under the License.
 
 from typing import List
-
+from adwp_base.errors import raise_AdwpEx as err
 from rest_framework.exceptions import ValidationError
 
 from rbac.models import Role, RoleTypes
 from rbac.utils import update_m2m_field
 
 
+def set_parametrized_from_child(role, children: List[Role]):
+    param_list = []
+
+    for child in children:
+        child_params = child.parametrized_by_type
+        for child_param in child_params:
+            if not check_child_parametrized(child_param, param_list):
+                param_list.append(child_param)
+    role.parametrized_by_type = param_list
+    role.save()
+
+
+def check_child_parametrized(child_param, param_list) -> bool:
+    cluster_hierarchy = {'cluster', 'service', 'component'}
+    if ((child_param == 'provider') and bool(cluster_hierarchy & set(param_list))) or (
+        (child_param in cluster_hierarchy) and ('provider' in param_list)
+    ):
+        msg = {'This children parametrized by types from different hierarchy'}
+        raise err('ROLE_CREATE_ERROR', msg)
```

Comment:
I think the ROLE_ERROR error will be better, since the function is called not only in the create method

Reply:
Fixed